tests/pipelines/test_pipelines_common.py

```diff
@@ -1808,6 +1809,77 @@ def callback_increase_guidance(pipe, i, t, callback_kwargs):
         # accounts for models that modify the number of inference steps based on strength
         assert pipe.guidance_scale == (inputs["guidance_scale"] + pipe.num_timesteps)
 
+    def test_official_callback(self):
```

Comment:
Maybe we just test the officially supported callback here?

Reply:
can I know here if the pipeline being tested is SD 1.5 or SDXL?

Reply:
Oh okay. So, IIUC, you're proposing not to add tests for the official callbacks then?

Reply:
Not quite, it's more like asking for guidance on how to do it following the rest of the code base, so maybe I'll split the questions:

- Are we're going to make tests for all the official callbacks (I think there's going to be at least 10), some for IP Adapters, SD, SDXL and probably Pixart and SD3. So it's not that I was proposing not to add tests, just adding more context to make the decision.
- For example, if I want to test the SDXL one, I didn't see any other test that checked the pipeline class. So what would be the best option for diffusers here? create only one test in `StableDiffusionXLPipelineFastTests`?, on all of them, create a mixin or just write it in the common one? This will be relevant also for the SD 1.5 but not for IP Adapters since they already have a mixin.

Reply:
That's what I would have done so I like it, I'll do the other tests then.

Reply:
I agree here; we can make a mixin and add to relevant pipeline tests. I think we should at least test SD and SDXL-based pipelines, it will be nice to have more comprehensive coverage to include Kandinsky, pixart, cascade

we also need to figure out a way to find out which callback works with which pipelines 

Here in my from_pipe tester mixin, I use this simple logic to figure out which ones are SD-based and which ones are SDXL based, https://github.com/huggingface/diffusers/blob/e0e8c58f64c024e89d125ddb563d0de4cf252dc1/tests/pipelines/test_pipelines_common.py#L626C9-L626C32